src/metadata_internal.h

```diff
@@ -0,0 +1,43 @@
+/* createrepo_c - Library of routines for manipulation with repodata
+ * Copyright (C) 2013  Tomas Mlcoch
+ *
+ * This program is free software; you can redistribute it and/or
+ * modify it under the terms of the GNU General Public License
+ * as published by the Free Software Foundation; either version 2
+ * of the License, or (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program; if not, write to the Free Software
+ * Foundation, Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301,
+ * USA.
+ */
+
+#ifndef __C_CREATEREPOLIB_METADATA_INTERNAL_H__
+#define __C_CREATEREPOLIB_METADATA_INTERNAL_H__
```

Comment:
Is this file supposed to have more than just modulemd stuff? Because if not, this is probably the wrong name for the file...

Reply:
I named it based on the conventions for other files. It can theoretically have any private internal function definitions needed by createrepolib_metadata. Currently we only need it for module stuff.